akka-docs/src/main/paradox/discovery/index.md

```diff
@@ -0,0 +1,245 @@
+
+Java
+:  @@snip [CompileOnlyTest.java](/akka-discovery/src/test/java/jdoc/akka/discovery/CompileOnlyTest.java) { #basic }
+
+
+`portName` and `protocol` are optional and their meaning is interpreted by the method.
+
+Scala
+:  @@snip [CompileOnlySpec.scala](/akka-discovery/src/test/scala/doc/akka/discovery/CompileOnlySpec.scala) { #full }
+
+Java
+:  @@snip [CompileOnlyTest.java](/akka-discovery/src/test/java/jdoc/akka/discovery/CompileOnlyTest.java) { #full }
+
+Port can be used when a service opens multiple ports e.g. a HTTP port and an Akka remoting port.
+
+## Discovery Method: DNS
+
+DNS discovery maps `Lookup` queries as follows:
+
+* `serviceName`, `portName` and `protocol` set: SRV query in the form: `_port._protocol._name` Where the `_`s are added.
+* Any query  missing any of the fields is mapped to a A/AAAA query for the `serviceName`
+
+The mapping between Akka service discovery terminology and SRV terminology:
+
+* SRV service = port
+* SRV name = serviceName
+* SRV protocol = protocol
+
+Configure `akka-dns` to be used as the discovery implementation in your `application.conf`:
+
+@@snip[application.conf](/akka-discovery/src/test/scala/akka/discovery/dns/DnsDiscoverySpec.scala){ #configure-dns }
+
+From there on, you can use the generic API that hides the fact which discovery method is being used by calling::
+
+Scala
+:   ```scala
+    import akka.discovery.ServiceDiscovery
+    val system = ActorSystem("Example")
+    // ...
+    val discovery = ServiceDiscovery(system).discovery
+    val result: Future[Resolved] = discovery.lookup("service-name", resolveTimeout = 500 milliseconds)
+    ```
+
+Java
+:   ```java
+    import akka.discovery.ServiceDiscovery;
+    ActorSystem system = ActorSystem.create("Example");
+    // ...
+    SimpleServiceDiscovery discovery = ServiceDiscovery.get(system).discovery();
+    Future<SimpleServiceDiscovery.Resolved> result = discovery.lookup("service-name", Duration.create("500 millis"));
+    ```
+
+### How it works
+
+DNS discovery will use either A/AAAA records or SRV records depending on whether a `Simple` or `Full` lookup is issued..
+The advantage of SRV records is that they can include a port.
+
+#### SRV records
+
+Lookups with all the fields set become SRV queries. For example:
+
+```
+dig srv service.tcp.akka.test
+
+; <<>> DiG 9.11.3-RedHat-9.11.3-6.fc28 <<>> srv service.tcp.akka.test
+;; global options: +cmd
+;; Got answer:
+;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 60023
+;; flags: qr aa rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 1, ADDITIONAL: 5
+
+;; OPT PSEUDOSECTION:
+; EDNS: version: 0, flags:; udp: 4096
+; COOKIE: 5ab8dd4622e632f6190f54de5b28bb8fb1b930a5333c3862 (good)
+;; QUESTION SECTION:
+;service.tcp.akka.test.         IN      SRV
+
+;; ANSWER SECTION:
+service.tcp.akka.test.  86400   IN      SRV     10 60 5060 a-single.akka.test.
+service.tcp.akka.test.  86400   IN      SRV     10 40 5070 a-double.akka.test.
```

Comment:
Do we want to add underscores here as well?

Reply:
Yep